k2/csrc/array.h

```diff
@@ -410,10 +424,34 @@ class Array2 {
     that
      it will have a different memory layout than the input.
   */
-  Array2<T> To(ContextPtr ctx) {
-    // TODO
-    Array2<T> array;
-    return array;
+  Array2<T> To(ContextPtr ctx) const {
+    if (dim0_ * dim1_ == 0) {
+      if (ctx->IsCompatible(*Context())) return *this;
+      Array2 ans(ctx, dim0_, dim1_);
+      return ans;
+    }
+
+    // clang-format off
+    if (ctx->IsCompatible(*Context()) &&
+          elem_stride0_ == dim1_)
```

Comment:
actually we shouldn't have to check that elem_stride_ == dim1_.  You can remove the documentation part that says the result will always be contiguous.

Reply:
Done.

Reply:
may be better if adding documentation to say we may make it contiguous if their cotext are not compatiable, as we called ToContiguous in that case.

Reply:
I will add it in the next pullrequest.

Reply:
Fixed in https://github.com/danpovey/k2/pull/146